logfire/_internal/integrations/httpx.py

```diff
@@ -44,14 +97,105 @@ def instrument_httpx(
+                request_hook = make_capture_request_headers_hook(request_hook)
+
+        if capture_response_headers:
+            if isinstance(client, httpx.AsyncClient):
+                response_hook = cast('AsyncResponseHook | None', response_hook)
+                response_hook = make_capture_async_response_headers_hook(response_hook)
+            else:
+                response_hook = cast('ResponseHook | None', response_hook)
+                response_hook = make_capture_response_headers_hook(response_hook)
+
+        tracer_provider = final_kwargs['tracer_provider']
+        instrumentor.instrument_client(client, tracer_provider, request_hook, response_hook)
+
+
+def make_capture_response_headers_hook(hook: ResponseHook | None) -> ResponseHook:
+    def capture_response_headers_hook(span: Span, request: RequestInfo, response: ResponseInfo) -> None:
+        capture_response_headers(span, request, response)
+        run_hook(hook, span, request, response)
+
+    return capture_response_headers_hook
+
+
+def make_capture_async_response_headers_hook(hook: AsyncResponseHook | None) -> AsyncResponseHook:
+    async def capture_response_headers_hook(span: Span, request: RequestInfo, response: ResponseInfo) -> None:
+        capture_response_headers(span, request, response)
+        await run_async_hook(hook, span, request, response)
+
+    return capture_response_headers_hook
+
+
+def make_capture_request_headers_hook(hook: RequestHook | None) -> RequestHook:
+    def capture_request_headers_hook(span: Span, request: RequestInfo) -> None:
+        capture_request_headers(span, request)
+        run_hook(hook, span, request)
+
+    return capture_request_headers_hook
+
+
+def make_capture_async_request_headers_hook(hook: AsyncRequestHook | None) -> AsyncRequestHook:
+    async def capture_request_headers_hook(span: Span, request: RequestInfo) -> None:
+        capture_request_headers(span, request)
+        await run_async_hook(hook, span, request)
+
+    return capture_request_headers_hook
+
+
+async def run_async_hook(hook: Callable[P, Any] | None, *args: P.args, **kwargs: P.kwargs) -> None:
+    if hook:
+        result = hook(*args, **kwargs)
+        while inspect.isawaitable(result):
+            result = await result
+
+
+def run_hook(hook: Callable[P, Any] | None, *args: P.args, **kwargs: P.kwargs) -> None:
+    if hook:
+        hook(*args, **kwargs)
+
+
+def capture_response_headers(span: Span, request: RequestInfo, response: ResponseInfo) -> None:
+    capture_headers(span, cast('httpx.Headers', response.headers), 'response')
+
+
+def capture_request_headers(span: Span, request: RequestInfo) -> None:
+    capture_headers(span, cast('httpx.Headers', request.headers), 'request')
+
+
+def capture_headers(span: Span, headers: httpx.Headers, request_or_response: Literal['request', 'response']) -> None:
+    span.set_attributes(
+        {
+            f'http.{request_or_response}.header.{header_name}': headers.get_list(header_name)
```

Comment:
@alexmojaki note that we comply with the OTel specification here, but the Python OTel itself doesn't comply, and they actually replace `-` by `_`.